internal/lines.go

```diff
@@ -0,0 +1,100 @@
+package internal
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type LineHandler struct {
+	Reporter      Reporter
+	BatchSize     int
+	MaxBufferSize int
+	FlushTicker   *time.Ticker
+	Format        string
+	failures      int64
+	mtx           sync.Mutex
+	lines         []string
+}
+
+func (lh *LineHandler) Start() {
+	go func() {
```

Comment:
If it were me, I would let this goroutine take care of the buffering as well as the flushing. You could eliminate the ```lines``` buffer and make it a ```chan(string)``` instead. Create it with your buffer size, and you have a goroutine-safe way of handling all of this with a lot less code.